Sources/NIO/EventLoopFuture.swift

```diff
@@ -859,16 +859,77 @@ extension EventLoopFuture {
     ///     - eventLoop: The `EventLoop` on which the new `EventLoopFuture` callbacks will fire.
     /// - returns: A new `EventLoopFuture`.
     public static func andAll(_ futures: [EventLoopFuture<Void>], eventLoop: EventLoop) -> EventLoopFuture<Void> {
-        let p0: EventLoopPromise<Void> = eventLoop.newPromise()
-        guard futures.count > 0 else {
-            p0.succeed(result: ())
-            return p0.futureResult
+        let body: EventLoopFuture<Void> = EventLoopFuture<Void>.reduce((), futures, eventLoop: eventLoop) {
+            (_: Void, _: Void) in
+            return ()
```

Comment:
I'm a little disappointed that even the first one is ;-) It should just be `{ _ in () }`.

Reply:
@moiseev we had bugs just blanket `{ _ in ... }`s because people thought it's some uninteresting type `()` and just ignored important things. In synchronous programming you also don't write `_ = foo()` all that often as most of the time you shouldn't ignore what's returned.
Therefore we mandated to explicitly write the types of the stuff you ignore just to make it obvious. In the `Void` case it should however be `()`. That makes it totally obvious that something unimportant is ignored and if that should change we'll get a compile-time error.

@karim-elngr maybe just change this to 

    let body: EventLoopFuture<Void> = EventLoopFuture<Void>.reduce((), futures, eventLoop: eventLoop) { ((), ()) in
        ()
    }

Reply:
This doesn't compile in swift-4.0.3 with the following error "Closure tuple parameter does not support destructuring"
```Swift
let body = EventLoopFuture<Void>.reduce((), futures, eventLoop: eventLoop) { ((), ()) in
    ()
}
```

Xcode even suggests the following fix:
```Swift
let body = EventLoopFuture<Void>.reduce((), futures, eventLoop: eventLoop) { (arg0, arg1) in
    let () = arg1
    let () = arg0
    ()
}
```

Even this
```Swift
{ _ in () }
```
Gives the following error "Contextual closure type '(Void, _) -> Void' expects 2 arguments, but 1 was used in closure body"

Doesn't compile unless we add an additional `_` because of [SE-0110](https://github.com/apple/swift-evolution/blob/master/proposals/0110-distingish-single-tuple-arg.md):
```Swift
{ _, _ in () }
```